planner/core/find_best_task.go

```diff
@@ -359,6 +362,34 @@ func (op *physicalOptimizeOp) appendCandidate(lp LogicalPlan, pp PhysicalPlan, p
 	pp.appendChildCandidate(op)
 }
 
+func (op *physicalOptimizeOp) appendPlanCostDetail(pp PhysicalPlan) *physicalOptimizeOp {
+	if op == nil || op.tracer == nil {
+		return nil
+	}
+	if op.currCostDetail != nil {
+		op.tracer.PhysicalPlanCostDetails[pp.ID()] = op.currCostDetail
+		op.currCostDetail = nil
+	}
+	op.currCostDetail = tracing.NewPhysicalPlanCostDetail(pp.ID(), pp.TP())
+	return op
+}
+
+func (op *physicalOptimizeOp) appendParamCostDetail(k string, v interface{}) *physicalOptimizeOp {
```

Comment:
I think this would be more consistent with `appendCurrCostDesc` below.
```suggestion
func (op *physicalOptimizeOp) appendCurrCostParam(k string, v interface{}) *physicalOptimizeOp {
```